hail_search/hail_search_query.py

```diff
@@ -250,6 +250,40 @@ def _missing_entry(entry):
         entry_type = dict(**entry.dtype)
         return hl.struct(**{k: hl.missing(v) for k, v in entry_type.items()})
 
+    def _filter_annotated_table(self, frequencies=None, **kwargs):
+        self._filter_by_frequency(frequencies)
+
+    def _filter_by_frequency(self, frequencies):
+        frequencies = {k: v for k, v in (frequencies or {}).items() if k in self.POPULATIONS}
+        if not frequencies:
+            return
+
+        for pop, freqs in sorted(frequencies.items()):
+            pop_filters = []
+            pop_expr = self._ht[self.POPULATION_FIELDS.get(pop, pop)]
+            pop_config = self._format_population_config(self.POPULATIONS[pop])
+            if freqs.get('af') is not None:
+                af_field = pop_config.get('filter_af') or pop_config['af']
+                pop_filters.append(pop_expr[af_field] <= freqs['af'])
+            elif freqs.get('ac') is not None:
+                ac_field = pop_config['ac']
+                if ac_field:
+                    pop_filters.append(pop_expr[ac_field] <= freqs['ac'])
+
+            if freqs.get('hh') is not None:
+                hom_field = pop_config['hom']
+                hemi_field = pop_config['hemi']
+                if hom_field:
+                    pop_filters.append(pop_expr[hom_field] <= freqs['hh'])
+                if hemi_field:
+                    pop_filters.append(pop_expr[hemi_field] <= freqs['hh'])
+
+            if pop_filters is not None:
+                pop_filter = pop_filters[0]
```

Comment:
The condition for the if statement should be `len(pop_filters) > 0`. Otherwise, `pop_filters[0]` can be out of range error.

Reply:
`len(None)` throws an error though. But yes, this could be cleaned up